### ISSUE TITLE
[RFE] Add variants for some kola tests that start from an empty rootfs

### ISSUE THREAD

**Issue by pothos** [flatcar/Flatcar#559]
## Current situation

Kola tests run from the release image as it is published, which includes a pre-populated rootfs. These rootfs contents change from release to release, while users stick with the initial copy when they installed their machines, or even start from an empty rootfs when the Ignition directive is used to setup a new rootfs. There are systemd-tmpfile rules in place that should ensure that users get up-to-date rootfs entries where needed, but we only have few kola tests that reformat the rootfs.

## Impact

When doing changes, in particular where software needs to get downstream patches to store config files in /usr or to enforce file creation with systemd-tmpfiles rules, we can easily miss this, causing problems for users that update from an older version or users that have to reformat the rootfs.

## Ideal future situation

A meaningful set of kola tests gets a second variant which adds the Ignition directive to reformat the rootfs.

## **Implementation options

The injection of the directive can be done in `RenderUserData()` and does not need to be duplicated per test. A copy of the test definition is enough with a test flag set to reformat the rootfs.

## Additional information

We have [docs for the Ignition directive](https://www.flatcar-linux.org/docs/latest/provisioning/cl-config/examples/#filesystems) and even docs with a command to [reset the rootfs on a provisioned instance](https://www.flatcar-linux.org/docs/latest/reference/developer-guides/sdk-disk-partitions/#stateful-root) (not needed for kola)

**Comment by tormath1** [comment]
@pothos thanks for the issue - do you have some example of failures ? (Maybe this one: https://github.com/flatcar-linux/Flatcar/issues/544#issuecomment-967199161)

> A meaningful set of kola tests gets a second variant which adds the Ignition directive to reformat the rootfs.

What about enforcing this behavior to all tests ? Except for the update one.

**Comment by pothos** [comment]
> What about enforcing this behavior to all tests ? Except for the update one.

We want to test both scenarios, the pre-populated rootfs we ship on the release image, and the empty rootfs.

Examples are when an unmodified Gentoo package would install something to `/etc/` and expect it to exist which will fail if there are no tmpfile directives symlinking to something under `/usr/` (the other alternative is a downstream patch to not use `/etc/` in the first place).
(Edit: The concrete example was logrotate but it's a tricky/not-fitting case since the service was timer-triggered it wouldn't show up in the failed services check immediately but only after a day or so, and the service even failed already with the pre-populated rootfs because there was a downstream patch that used wrong paths.)

**Comment by pothos** [comment]
@tormath1 Or I have another idea: We can ship the image with an empty rootfs by default, the few directories and symlinks we need get created on first boot anyway. This way we can't accidentally introduce a file on rootfs that older machines don't have because both old machines and new ones need to have the file defined through the content of the `/usr` partition (e.g., either a `/usr` path is used instead of `/etc` or a systemd-tmpfile rule exists to set it up on `/etc`),

**Comment by pothos** [comment]
Done as part of the /etc overlay feature.
